eureka2-core/src/main/java/com/netflix/eureka2/interests/Index.java

```diff
@@ -77,20 +78,20 @@ public void call(Subscriber<? super ChangeNotification<T>> subscriber) {
                 // Because new instance holder updates will be added while we replay them, they will be
                 // partially visible by the subscriber. When we replay buffered real time updates,
                 // they may overlap with what was already sent from the init holder.
-                NotificationsSubject<T> realTimeSubject = NotificationsSubject.create();
+                // TODO can we make this buffering cheaper?
+                final NotificationsSubject<T> realTimeSubject = NotificationsSubject.create();
                 realTimeSubject.pause();
                 realTimeSource.subscribe(realTimeSubject);
-                realTimeSubject.subscribe(subscriber);
 
-                // Reply initial state
-                for (ChangeNotification<T> notification : initStateHolder) {
-                    subscriber.onNext(notification);
-                }
-
-                // Now open real time source
-                realTimeSubject.resume();
+                realTimeSubject.mergeWith(Observable.from(initStateHolder).doOnCompleted(new Action0() {
```

Comment:
we need to merge here instead of concat (which would be Ob.from(init).concatWith(realTimeSubject) ) as we need to ensure that the realTimeSubject is subscribed to before resume() is called on it. Otherwise, we would lose buffered messages that are stored while paused.